user-operator/src/main/java/io/strimzi/operator/user/UserOperatorConfig.java

```diff
@@ -14,11 +13,106 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Properties;
+import java.util.HashMap;
+import java.util.Objects;
+import java.util.Set;
+import java.util.Collections;
 
 /**
  * Cluster Operator configuration
  */
 public class UserOperatorConfig {
+
+
+    private static abstract class Type<T> {
+        abstract T parse(String s);
+    }
+
+    /**
+     * A java string
+     */
+    private static final Type<? extends String> STRING = new Type<>() {
+        @Override
+        public String parse(String s) {
+            return s;
+        }
+    };
+
+    /**
+     * A java Long
+     */
+    private static final Type<? extends Long> LONG = new Type<>() {
+        @Override
+        public Long parse(String s) {
+            return Long.parseLong(s);
+        }
+    };
+
+    /**
+     * A Java Integer
+     */
+    private static final Type<? extends Integer> INTEGER = new Type<>() {
+        @Override
+        Integer parse(String s) {
+            return Integer.parseInt(s);
+        }
+    };
+
+    /**
+     * A Java Boolean
+     */
+    private static final Type<? extends Boolean> BOOLEAN = new Type<>() {
+        @Override
+        Boolean parse(String s) {
+            return Boolean.parseBoolean(s);
+        }
+    };
+
+    /**
+     * A kubernetes selector.
+     */
+    private static final Type<? extends Labels> LABEL_PREDICATE = new Type<>() {
+        @Override
+        public Labels parse(String s) {
+            return Labels.fromString(s);
+        }
+    };
+
+    /** Wrapper class for all the configuration parameters */
```

Comment:
Document the _concept_, or what this class is trying to model. "Wrapper class" is not really a concept. 

e.g.

```
Models a configuration parameter, identified by a unique key, which may be required, and if not may have a default value. Optional parameters without a default value implicitly have a null default. The key is also the name of the environment variable from which the value may be read, when it is read from the environment.
```

That explains to someone coming along later what the intent for this class is.